src/gathernodeinputhandler.cpp

```diff
@@ -15,27 +15,49 @@
 //*****************************************************************************
 #include "gathernodeinputhandler.hpp"
 
+#include <functional>
+
 #include "logging.hpp"
+#include "nodesessionmetadata.hpp"
 #include "ov_utils.hpp"
+#include "tensorinfo.hpp"
 
 namespace ovms {
 
-GatherNodeInputHandler::GatherNodeInputHandler(uint32_t inputsMissingCount, session_id_t shardsCount) :
-    NodeInputHandler(inputsMissingCount) {
-    remainingDependencies *= shardsCount;
+GatherNodeInputHandler::GatherNodeInputHandler(uint32_t inputsMissingCount, const CollapsingDetails& collapsingDetails) :
+    NodeInputHandler(inputsMissingCount),
+    collapsingDetails(std::make_unique<CollapsingDetails>(collapsingDetails)) {
+    remainingDependencies = std::accumulate(
+        collapsingDetails.collapsedSessionSizes.begin(),
+        collapsingDetails.collapsedSessionSizes.end(),
+        remainingDependencies,
+        std::multiplies<session_id_t>());
 }
 
-void GatherNodeInputHandler::setInput(const std::string& inputName, InferenceEngine::Blob::Ptr& ptr, session_id_t shardId) {
-    auto it = shardsStorage.find(inputName);
-    if (it == shardsStorage.end()) {
+Status GatherNodeInputHandler::setInput(const std::string& inputName, InferenceEngine::Blob::Ptr& ptr, session_id_t shardId) {
+    auto inputsShardsIt = shardsStorage.find(inputName);
+    if (inputsShardsIt == shardsStorage.end()) {
         shard_map_t shardMap{{shardId, ptr}};
         auto itDidInsertPair = shardsStorage.emplace(inputName, std::move(shardMap));
         if (!itDidInsertPair.second) {
             throw std::runtime_error("Tried to insert the same input twice with the same shard id");
         }
     } else {
-        it->second.emplace(shardId, ptr);
+        auto firstShardTensor = inputsShardsIt->second.begin()->second;
+        if (firstShardTensor->getTensorDesc() != ptr->getTensorDesc()) {
+            SPDLOG_LOGGER_ERROR(dag_executor_logger, "Shard: {} tensor description differ. First shard desc: {}, current shard desc: {}",
+                shardId,
+                TensorInfo("firstShard", firstShardTensor->getTensorDesc()).getPrintableString(),
```

Comment:
Wouldn't `.string()` be better name? Or maybe use implicit/explicit to string cast operator?

Reply:
I don't think you need the conversion from OV TensorDesc to OVMS TensorInfo just to print tensordesc data. Why not to directly implement OV TensorDesc to string operator?